### PR TITLE
Metrics - use second resolution by default 

### PR DESCRIPTION
**Pull request by gsps1**
the REST API will return 1 sec resolution by default.
The REST API will adjust resolution automatically if you specify resolution=auto param

### CODE REVIEW

**Comment by abaranau** [comment]
@awholegunch none at this point. It behaves differently in new UI only

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java [hunk 1]
```diff
@@ -414,17 +411,18 @@ private static void parseTimeseries(Map<String, List<String>> queryParams, Metri
...
-        // determine resolution, based on difference.
-        MetricsRequest.TimeSeriesResolution autoResolution = getResolution(endTime - startTime);
-        builder.setTimeSeriesResolution(autoResolution.getResolution());
-        resolution = autoResolution.getResolution();
-      }
-      if (queryParams.containsKey(COUNT)) {
-        count = Integer.parseInt(queryParams.get(COUNT).get(0));
+      if (queryParams.containsKey(RESOLUTION)) {
+        if (queryParams.get(RESOLUTION).get(0).equals("auto")) {
+          //auto determine resolution, based on time difference.
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: space after "//"

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java [hunk 2]
```diff
@@ -414,17 +411,18 @@ private static void parseTimeseries(Map<String, List<String>> queryParams, Metri
...
       endTime = TimeMathParser.parseTime(now, queryParams.get(END_TIME).get(0));
-      if (!queryParams.containsKey(RESOLUTION)) {
-        // determine resolution, based on difference.
-        MetricsRequest.TimeSeriesResolution autoResolution = getResolution(endTime - startTime);
-        builder.setTimeSeriesResolution(autoResolution.getResolution());
-        resolution = autoResolution.getResolution();
-      }
-      if (queryParams.containsKey(COUNT)) {
-        count = Integer.parseInt(queryParams.get(COUNT).get(0));
+      if (queryParams.containsKey(RESOLUTION)) {
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: shouldn't it be "if (!queryParams.containsKey(RESOLUTION) || queryParams.get(RESOLUTION).get(0).equals("auto"))

Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: also, please extract "auto" into a constant, at least in this class.

Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: also, would help to make code readable: extract methods like isAutoResolution(queryParams) or similar

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java [hunk 3]
```diff
@@ -414,17 +411,18 @@ private static void parseTimeseries(Map<String, List<String>> queryParams, Metri
...
+        if (queryParams.get(RESOLUTION).get(0).equals("auto")) {
+          //auto determine resolution, based on time difference.
+          MetricsRequest.TimeSeriesResolution autoResolution = getResolution(endTime - startTime);
+          builder.setTimeSeriesResolution(autoResolution.getResolution());
+          resolution = autoResolution.getResolution();
+        }
       } else {
-        count = (int) (((endTime / resolution * resolution) - (startTime / resolution * resolution)) / resolution + 1);
+        builder.setTimeSeriesResolution(MetricsRequest.TimeSeriesResolution.SECOND.getResolution());
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: init resolution variable before this line and use it in builder.setTimeSeriesResolution(resolution) - will make code more readable

**Comment by abaranau** [comment]
@gsps1 couple comments on code style, good to merge after those are fixed

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java (hunk 2 above)
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: if (!queryParams.containsKey(RESOLUTION)), we want to use 1 second as resolution as default.

Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java]: OK

**Comment by abaranau** [comment]
good to merge?

**Comment by gsps1** [comment]
yeah. merging.

**Pull request merged**